services/auth/source/ldap/source_group_sync.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package ldap
+
+import (
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/log"
+)
+
+// SyncLdapGroupsToTeams maps LDAP groups to organization and team memberships
+func (source *Source) SyncLdapGroupsToTeams(user *models.User, ldapTeamAdd map[string][]string, ldapTeamRemove map[string][]string, orgCache map[string]*models.User, teamCache map[string]*models.Team) {
+	var err error
+	if source.TeamGroupMapRemoval {
+		// when the user is not a member of configs LDAP group, remove mapped organizations/teams memberships
+		removeMappedMemberships(user, ldapTeamRemove, orgCache, teamCache)
+	}
```

Comment:
What if `TeamGroupMapRemoval==true` and `TeamGroupMapEnabled==false` here?

Allowing setting `TeamGroupMapRemoval` and `TeamGroupMapEnabled` separately seems fragile.

Reply:
> What if TeamGroupMapRemoval==true and TeamGroupMapEnabled==false here?

In this case the behavior is just like setting both to true, as TeamGroupMapRemoval is an extended edition of this feature, which also includes the base behavior i.e. adding to teams and orgs.
But I could improve the UI by disabling the `TeamGroupMapEnables` checkbox and set it to checked, if `TeamGroupMapRemoval` is enabled